File: exercism/java/armstrong-numbers/src/main/java/ArmstrongNumbers.java


class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int numberOfDigits = String.valueOf(numberToCheck).length();
        return String.valueOf(numberToCheck).chars().map(Character::getNumericValue).mapToLong(i -> (long) Math.pow(i, numberOfDigits)).sum() == numberToCheck;
    }

}

File: exercism/dart/armstrong-numbers/lib/armstrong_numbers.dart

class ArmstrongNumbers {
  bool isArmstrongNumber(String number) {
    return number
            .split('')
            .map((e) => BigInt.parse(e).pow(number.length))
            .reduce((value, element) => value + element)
            .toString() ==
        number;
  }
}
